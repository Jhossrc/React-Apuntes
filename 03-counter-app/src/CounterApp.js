import React from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
  return (
    <>
      <h1>CounterApp</h1>
      <hr/>
      <h2> <i>{value}</i> </h2>
    </>
  );
};

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;
