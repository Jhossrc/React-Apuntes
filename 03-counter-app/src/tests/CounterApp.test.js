import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Pruebas en el componente CounterApp', () => {
  test('CounterApp : Debe de mostar <CounterApp/> correctamente', () => {
    const wrapper = shallow(<CounterApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('CounterApp : Debe de mostrar el value por defecto de 100', () => {
    const valueTest = 100;
    const wrapper = shallow(<CounterApp value={valueTest} />);

    const valueText = wrapper.find('h2 i').text();

    expect(valueText.trim()).toBe(valueTest.toString());
  });
});
