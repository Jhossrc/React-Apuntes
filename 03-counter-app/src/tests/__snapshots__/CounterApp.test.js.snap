// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CounterApp CounterApp : Debe de mostar <CounterApp/> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp!!
  </h1>
  <hr />
  <h2>
    <i>
      0
    </i>
  </h2>
  <button
    onClick={[Function]}
  >
    Click +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Click -1
  </button>
</Fragment>
`;
